src/Containers.jsx

```diff
@@ -677,28 +677,27 @@ class Containers extends React.Component {
 
         // Convert to the search result output
         let localImages = null;
+        let nonIntermediateImages = null;
         if (this.props.images) {
-            localImages = Object.keys(this.props.images).reduce((images, id) => {
+            localImages = Object.keys(this.props.images).map(id => {
                 const img = this.props.images[id];
-                if (img.RepoTags) {
-                    img.Index = img.RepoTags[0].split('/')[0];
-                    img.Name = img.RepoTags[0];
-                    img.toString = function imgToString() { return this.Name };
-                    images.push(img);
-                }
-                return images;
+                img.Index = img.RepoTags ? img.RepoTags[0].split('/')[0] : "";
```

Comment:
`?:` doesn't work -- in JS, an empty list is truthy. It'd be more robust to just do `img?.RepoTags[0].split('/')[0]` or `img.RepoTags?.[0].split('/')[0]` depending on whether the whole RepoTags is missing for intermediate images or whether it's empty then.

Does the code depend on the empty string anywhere? Is that for type reasons? Leaving it undefined feels more obvious and more JS-y, too.

Reply:
This would require a lot more cleanup or even a helper:
```
                return this.props.images[a].isSystem ? 1 : -1;
            const name_a = this.props.images[a].RepoTags ? this.props.images[a].RepoTags[0] : "";
            const name_b = this.props.images[b].RepoTags ? this.props.images[b].RepoTags[0] : "";
```

Reply:
ok, so I figure you could just change the condition to `img.RepoTags?.[0] ? ...` then? That should cover all cases.